conans/test/functional/toolchains/cmake/test_cmake_toolchain.py

```diff
@@ -157,12 +157,8 @@ class Pkg(ConanFile):
             def generate(self):
                 # Get the toolchains from "tools.cmake.cmaketoolchain:user_toolchain" conf at the
                 # tool_requires
-                user_toolchains = []
-                for dep in self.dependencies.direct_build.values():
-                    ut = dep.conf_info["tools.cmake.cmaketoolchain:user_toolchain"]
-                    if ut:
-                        user_toolchains.append(ut.replace('\\\\', '/'))
-
+                user_toolchains = self.conf.get("tools.cmake.cmaketoolchain:user_toolchain", default=[], check_type=list)
```

Comment:
Nop. Just let the CMakeToolchain internals do the right thing, no longer necessary. You could even remove the ``generate()`` method, leave ``generators="CMakeToolchain"`` and it should work as expected.